assignment6/src/Parking.java

```diff
@@ -0,0 +1,107 @@
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.logging.Logger;
+
+import static java.util.Collections.nCopies;
+
+/**
+ * Parking class for rearranging cars
+ */
+public class Parking {
+    /**
+     * Return sequence of movement cars (from which places cars move to which).
+     * Function changes beginParking.
+     * <p>
+     * Complexity: O(n)
+     *
+     * @param beginParking -- sequence of cars on the beginning (from zero)
+     * @param endParking   -- sequence of cars on the end
+     * @return -- sequence of movements
+     */
+    public List<Movement> RearrangingCars(int[] beginParking, int[] endParking) {
+        List<Integer> beginCarsPosition = createCarsPositionList(beginParking);
+        List<Movement> movements = new LinkedList<>();
+
+        while (true) {
+            int posOfZero = beginCarsPosition.get(0);
+            int carToPosOfZero = endParking[posOfZero];
+            int posOfCarToZeroPos = beginCarsPosition.get(carToPosOfZero);
+
+            if (endParking[posOfZero] != 0) {
```

Comment:
Use `carToPosOfZero` instead of `endParking[posOfZero]`, since you've already put it in a variable.